tests/trace/test_evaluations.py

```diff
@@ -747,3 +780,36 @@ def function_score(image, dc, model, obj, text, model_output) -> bool:
     assert "table_query" in access_log
     assert "obj_read" in access_log
     assert "file_content_read" in access_log
+
+
+@pytest.mark.asyncio
+async def test_feedback_is_correctly_linked(client):
+    @weave.op
+    def predict(text: str) -> str:
+        return text
+
+    @weave.op
+    def score(text, model_output) -> bool:
+        return text == model_output
+
+    eval = weave.Evaluation(
+        dataset=[{"text": "hello"}],
+        scorers=[score],
+    )
+    res = await eval.evaluate(predict)
+    calls = client.server.calls_query(
+        tsi.CallsQueryReq(
+            project_id=client._project_id(),
+            include_feedback=True,
+            filter=tsi.CallsFilter(op_names=[get_ref(predict).uri()]),
+        )
+    )
+    assert len(calls.calls) == 1
+    assert calls.calls[0].summary["weave"]["feedback"]
+    feedbacks = calls.calls[0].summary["weave"]["feedback"]
+    assert len(feedbacks) == 1
+    feedback = feedbacks[0]
+    assert feedback["feedback_type"] == SCORE_TYPE_NAME
+    assert feedback["payload"]["name"] == "score"
+    assert feedback["payload"]["op_ref"] == get_ref(score).uri()
```

Comment:
nit: we have some inconsistencies in our code.  Some places use `get_ref(obj)`, whereas others use `obj.ref`

Reply:
i think `get_ref` and `set_ref` should be used everywhere, since `ref` is often tacked on in different ways based on the object and it seems more robust to just have helpers here.

Reply:
Are there other ways than `.ref`?  I'm not against using `get_ref`, but I do think we should be consistent

Reply:
we should probably standardize:
```

def get_ref(obj: Any) -> Optional[ObjectRef]:
    return getattr(obj, "ref", None)


def remove_ref(obj: Any) -> None:
    if get_ref(obj) is not None:
        if "ref" in obj.__dict__:  # for methods
            obj.__dict__["ref"] = None
        else:
            obj.ref = None


def set_ref(obj: Any, ref: Optional[Ref]) -> None:
    """Try to set the ref on "any" object.

    We use increasingly complex methods to try to set the ref
    to support different kinds of objects. This will still
    fail for python primitives, but those can't be traced anyway.
    """
    try:
        obj.ref = ref
    except:
        try:
            setattr(obj, "ref", ref)
        except:
            try:
                obj.__dict__["ref"] = ref
            except:
                raise ValueError(f"Failed to set ref on object of type {type(obj)}")
```